server/datastore/mysql/secret_variables.go

```diff
@@ -68,3 +68,80 @@ func (ds *Datastore) GetSecretVariables(ctx context.Context, names []string) ([]
 
 	return secretVariables, nil
 }
+
+func (ds *Datastore) ExpandEmbeddedSecrets(ctx context.Context, document string) (string, error) {
+	embeddedSecrets := fleet.ContainsPrefixVars(document, fleet.ServerSecretPrefix)
+
+	secrets, err := ds.GetSecretVariables(ctx, embeddedSecrets)
+	if err != nil {
+		return "", ctxerr.Wrap(ctx, err, "expanding embedded secrets")
+	}
+
+	secretMap := make(map[string]string, len(secrets))
+
+	for _, secret := range secrets {
+		secretMap[secret.Name] = secret.Value
+	}
+
+	missingSecrets := []string{}
+
+	for _, wantSecret := range embeddedSecrets {
+		if _, ok := secretMap[wantSecret]; !ok {
+			missingSecrets = append(missingSecrets, fmt.Sprintf("$FLEET_SECRET_%s", wantSecret))
+		}
+	}
+
+	if len(missingSecrets) > 0 {
+		return "", ctxerr.Errorf(ctx, "embedded secrets missing from datastore: %s", strings.Join(missingSecrets, ", "))
```

Comment:
Should use MissingSecretsError here?

Reply:
Yes it should, I probably wrote this before I created the error type

Reply:
Yes